### ISSUE TITLE
Can we finally remove Settings Rules?

### ISSUE THREAD

**Issue by john-science** [terrapower/armi#888]
For the past few years, the `SettingsRules` system has been antiquated at best, and redundant now that we have settings validators:

https://github.com/terrapower/armi/blob/f0d27e7405bde450b1ba01825c95783080974c53/armi/settings/settingsRules.py#L20-L22

Generally, we want to move away from "this code runs when you import ARMI" to things that more controllable and less magical.

The only place I can see in ARMI this will matter is is `armi/physics/neutronics/settings.py` there are some "rules" that will have to be converted to "validators". Which is a very small lift.

> But how many downstream projects would have to be converted first? That's what determines our level-of-effort here.

**Comment by jakehader** [comment]
Yes, let's remove! I think of this, the value that it can (if folks were to use it appropriately) is to identify specific functions/methods/settings that will be deprecated soon, but auto migration in this way leads to user never fixing inputs and it's then even more difficult to remove the setting because the auto migration gets relied upon.

**Comment by john-science** [comment]
I have reviewed as many downstream projects as I have access to, and none of them use `settingsRules`.

Happily, I believe that makes this change quite easy to implement.

**Comment by john-science** [comment]
Okay, so I gave this a first pass.  I moved the neutronics settings from `settingsRules` to `SettingValidator`s, and then went fine.

But when I tried to remove the (apparently unused) `settingsRules`, some unit tests broke, because `Setting.addOption()` doesn't work without `settingsRules`, somehow. Also, I believe our current `invalid settings` logic is tied up (extremely indirectly) in `settingsRules.py`.

So, that needs to be fixed to remove `settingsRules.py`. But even if I don't remove that, I'll want to fix these two point anyway. Because it appears we have two important features that are entirely undocumented and hidden.
